web-api/src/personalization/routes.rs

```diff
@@ -429,8 +436,26 @@ struct DevOption {
 }
 
 impl DevOption {
-    fn is_some(&self) -> bool {
-        self.hybrid.is_some() || self.max_number_candidates.is_some()
+    fn validate(&self, enable_dev: bool, count: usize) -> Result<(), Error> {
+        if !enable_dev && (self.hybrid.is_some() || self.max_number_candidates.is_some()) {
+            // notify the caller instead of silently discarding the dev option
+            return Err(ForbiddenDevOption::DevDisabled.into());
+        }
```

Comment:
:question:/:pushpin: This is quite prone to bugs when adding new dev options . Maybe we find a better way. But either way outside of this PR.

Reply:
there is probably no nice solution. we should remove dev options once they are not needed anymore (for example if we move a feature to production) to keep the complexity of the dev options low.